internal/action/run.go

```diff
@@ -0,0 +1,89 @@
+package action
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/sethvargo/go-githubactions"
+)
+
+// OutputString takes an interface and returns a stringified version
+func OutputString(i interface{}) (string, error) {
+	if i == nil {
+		return "", nil
+	}
+
+	t := reflect.TypeOf(i)
+	switch t.Kind() {
+	case
+		reflect.Array,
+		reflect.Map,
+		reflect.Slice,
+		reflect.Struct:
+		b, err := json.Marshal(i)
+		if err != nil {
+			return "", err
+		}
+		return string(b), nil
+
+	case
+		reflect.Chan,
```

Comment:
These don't really make sense, the code needs to cover the potential values of a Terraform Cloud output. An `interface{}` could be anything but it's only useful for have test coverage for the actual concrete types that are expected. An output's value can't be a channel or function.

Testing this could be done naively by mocking the server response. But the alternative is to reverse engineer, which should work fine in this case. Check for a few concrete types you know don't need encoding (actually 1, string), and anything else return JSON encoded.